docs/migrating.md

```diff
@@ -0,0 +1,57 @@
+# Migrating from v1alpha1 to v1alpha2
```

Comment:
How about having a separate folder for migrations doc instead, i.e `docs/migrations/v1alpha1-to-v1alpha2.md` so that we keep it separate and add new migration docs in the future following the above format?